src/Engine/Animation/AnimationController.cpp

```diff
@@ -71,14 +88,37 @@ void AnimationController::Load(const std::string& name) {
         }
     }
 
+    uint32_t numBools = boolMap.size();
+    file.read(reinterpret_cast<char*>(&numBools), sizeof(uint32_t));
+    boolMap.resize(numBools);
+    for (auto i = 0; i < numBools; ++i) {
+        file.read(reinterpret_cast<char*>(boolMap[i]), sizeof(BoolItem));
```

Comment:
The destination memory (`boolMap[i]` pointer) has not been allocated. Resizing the map has only reserved memory for the pointers themselves.